lettuce/options/base_options.py

```diff
@@ -1,206 +1,234 @@
-        )
-
-        self._parser.add_argument(
-            "--max_separation_ancestor",
-            type=int,
-            required=False,
-            default=1,
-            help="max separation ancestor",
-        )
-
-        self._parser.add_argument(
-            "--vector_search",
-            action=argparse.BooleanOptionalAction,
-            required=False,
-            default=True,
-            help="Try vector search before LLM?",
-        )
-
-        self._parser.add_argument(
-            "--use_llm",
-            action=argparse.BooleanOptionalAction,
-            required=False,
-            default=True,
-            help="Use LLM?",
-        )
-
-        self._parser.add_argument(
-                "--embedding-top-k",
-                type=int,
-                required=False,
-                default=5,
-                help="Number of suggestions to return from vector search for RAG."
-         )
-
-        self._initialized = True
-
-    def parse(self) -> argparse.Namespace:
-        """
-        Parses the arguments passed to the script
-
-        Parameters
-        ----------
-        None
-
-        Returns
-        -------
-        opt: argparse.Namespace
-            The parsed arguments
-        """
-        if not self._initialized:
-            self.initialize()
-        self._opt, _ = self._parser.parse_known_args()
-
-        args = vars(self._opt)
-        self._print(args)
-
-        return self._opt
-
-    def _print(self, args: Dict) -> None:
-        """
-        Prints the arguments passed to the script
-
-        Parameters
-        ----------
-        args: dict
-            The arguments to print
-
-        Returns
-        -------
-        None
-        """
+from pydantic_settings import BaseSettings, SettingsConfigDict
+from .pipeline_options import EmbeddingModelName
+
+class BaseOptions(BaseSettings):
+    model_config = SettingsConfigDict(
+            env_file=".env",
+            env_file_encoding="utf-8",
+            )
+    db_host: str = "localhost"
+    db_user: str = "postgres"
+    db_password: str = "password"
+    db_name: str = "omop"
+    db_port: int = 5432
+    db_schema: str = "cdm"
+    db_vectable: str = "embeddings"
+    # have a branch where this comes from database - hard to integrate,#TODO
+    db_vecsize: int = 384 
+    
+    llm_model: LLMModel = LLMModel.LLAMA_3_1_8B
+    temperature: float = 0.0
+
+    embedding_model: EmbeddingModelName = EmbeddingModelName.BGESMALL
+    embedding_top_k: int = 5
+    
+    auth_api_key: str | None = None
+
+    def connection_url(self) -> str:
+        return f"postgresql://{self.db_user}:{self.db_password}@{self.db_host}:{self.db_port}/{self.db_name}"
+
+    def print(self):
         print("------------ Options -------------")
-        for k, v in args.items():
+        for k, v in self.model_dump().items():
             print(f"{str(k)}: {str(v)}")
         print("-------------- End ---------------")
+
+
+
+# class BaseOptions:
```

Comment:
I think you've left this in, as it can be deleted instead of commented?

Otherwise - all lgtm